### PR TITLE
1836Jr30: Can only by one of each rank from the depot on a round

### PR DESCRIPTION
**Pull request by jenf**
Implements
A corporation may only buy one of each train type per operating turn

### CODE REVIEW

File: lib/engine/game/g_1836_jr30.rb [hunk 1]
```diff
@@ -15,6 +15,10 @@ class G1836Jr30 < Base
 
       SELL_BUY_ORDER = :sell_buy_sell
       TRACK_RESTRICTION = :permissive
+
+      def operating_round(round_num)
```
Comment (tobymao) [inline lib/engine/game/g_1836_jr30.rb:19]: why do you need this?

File: lib/engine/round/g_1836_jr_30/operating.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          @depot_trains_bought = []
+        end
+
+        def start_operating
+          @depot_trains_bought = []
+          super
+        end
+
+        def buyable_trains
+          super.reject { |x| x.from_depot? && @depot_trains_bought.include?(x.distance) }
```
Comment (tobymao) [inline lib/engine/round/g_1836_jr_30/operating.rb:20]: better to track by the sym, train.sym

File: lib/engine/round/g_1836_jr_30/operating.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        end
+
+        def can_buy_train?
+          super && (@depot_trains_bought.empty? || buyable_trains.any?)
+        end
+
+        def process_buy_train(action)
+          # Since the train won't be in the depot after being bougth store the state now.
+          add_to_list = action.train.from_depot?
+          super
```
Comment (tobymao) [inline lib/engine/round/g_1836_jr_30/operating.rb:30]: can't we do

@depoat_trains_bought << action.train.sym if action.train.from_depot?
super

Comment (jenf) [reply lib/engine/round/g_1836_jr_30/operating.rb:30]: nope, since process_buy_trains calls buyable_trains and means no train can be bought

**Pull request merged**